interface/src/app/sample_annotation/annotation.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * "adage.sampleAnnotation" module.
+ */
+
+angular.module('adage.sampleAnnotation', [
+  'ui.router',
+  'ui.bootstrap',
+  'adage.utils'
+])
+
+.config(['$stateProvider', function($stateProvider) {
+  $stateProvider.state('sampleAnnotation', {
+    url: '/sample_annotation?node&samples',
+    views: {
+      main: {
+        templateUrl: 'sample_annotation/annotation.tpl.html',
+        controller: 'SampleAnnotationCtrl as ctrl'
+      }
+    },
+    data: {pageTitle: 'Sample Annotations'}
+  });
+}])
+
+.factory('ActivityService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
+    return $resource(ApiBasePath + 'activity');
+  }
+])
+
+.factory('SampleService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
+    return $resource(ApiBasePath + 'sample');
+  }
+])
+
+.controller('SampleAnnotationCtrl', [
+  '$stateParams', '$q', '$log', 'errGen', 'ActivityService', 'SampleService',
+  function($stateParams, $q, $log, errGen, ActivityService, SampleService) {
+    var self = this;
+    self.queryStatus = 'Connecting to the server ...';
+
+    // Ensure that the URL includes sample ID(s).
+    if (!$stateParams.samples) {
+      self.queryStatus = 'Please specify sample ID(s) in the URL.';
+      return;
+    }
+
+    var samplesInUrl = $stateParams.samples;
+    self.uniqueAnnotationTypes = [];
+    // Note: Ideally self.uniqueAnnotationTypes should be a hashtable (such as
+    // an object of Set) instead of Array, but Set is only available since ES6.
+
+    self.hasNode = false;
+    var sampleDict = {};
+
+    // Function that sets valid sample IDs asynchronously:
+    var getSampleIDPromise = function() {
+      var deferred = $q.defer();
+      // If 'node' is found in the URL, get the activity values based on node
+      // and sample(s):
+      if ($stateParams.node) {
+        self.hasNode = true;
+        self.precision = 5;
```

Comment:
The main reason this variable is defined here is because it is actually the precision of activity values. So it is only needed when `node` parameter is in the URL. Having it defined here will save a few bytes of memory when this parameter is not in the URL.